scripts/grading/grade_submission.py

```
@@ -611,24 +628,157 @@ def _normalize_name(name: str) -> str:
+                    {"role": "system", "content": RUBRIC_SYSTEM},
+                    {"role": "user", "content": prompt},
+                ],
+            )
+            raw = _resp.choices[0].message.content.strip()
+        else:
+            print(
+                f"WARNING: ai_extract_rubric_criteria does not support provider '{provider}' — "
+                "returning no criteria. Supported providers: anthropic, openai."
+            )
+            return []
+    except Exception as exc:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)